packages/common/dynamoFunctions/src/tokenMapping.ts

```diff
@@ -0,0 +1,164 @@
+
+interface UserDetails {
+  family_name: string,
+  given_name: string
+}
+
+interface TokenMappingItem {
+    username: string,
+    cis2AccessToken?: string,
+    cis2RefreshToken?: string,
+    cis2ExpiresIn?: string,
+    cis2IdToken?: string,
+    apigeeAccessToken?: string,
+    apigeeRefreshToken?: string,
+    apigeeExpiresIn?: number,
+    selectedRoleId?: string,
+    userDetails?: UserDetails,
+    rolesWithAccess?: Array<RoleDetails>,
+    rolesWithoutAccess?: Array<RoleDetails>
+    currentlySelectedRole?: RoleDetails
+  }
+
+export const insertTokenMapping = async (
+  documentClient: DynamoDBDocumentClient,
+  tokenMappingTableName: string,
+  item: TokenMappingItem,
+  logger: Logger
+): Promise<void> => {
+  logger.debug("Inserting into tokenMapping", {item, tokenMappingTableName})
+  try {
+    await documentClient.send(
+      new PutCommand({
+        TableName: tokenMappingTableName,
+        Item: item
+      })
+    )
+    logger.debug("Successfully inserted into tokenMapping", {tokenMappingTableName})
+  } catch(error) {
+    logger.error("Error inserting into tokenMapping", {error})
+    throw new Error("Error inserting into tokenMapping")
+  }
+}
+
+export const updateTokenMapping = async (
+  documentClient: DynamoDBDocumentClient,
+  tokenMappingTableName: string,
+  tokenMappingItem: TokenMappingItem,
+  logger: Logger
+): Promise<void> => {
+  const currentTime = Math.floor(Date.now() / 1000)
+
+  logger.debug("Updating data in tokenMapping", {tokenMappingItem, tokenMappingTableName})
+
+  try {
+    let expiryTimestamp
+    if (tokenMappingItem.apigeeExpiresIn) {
+      expiryTimestamp = currentTime + Number(tokenMappingItem.apigeeExpiresIn)
+    }
+
+    const expressionParts: Array<string> = []
+    const expressionAttributeValues: Record<string, unknown> = {}
+
+    const optionalFields: Array<[keyof typeof tokenMappingItem, unknown]> = [
+      ["cis2AccessToken", tokenMappingItem.cis2AccessToken],
+      ["cis2RefreshToken", tokenMappingItem.cis2RefreshToken],
+      ["cis2ExpiresIn", tokenMappingItem.cis2ExpiresIn],
+      ["cis2IdToken", tokenMappingItem.cis2IdToken],
+      ["apigeeAccessToken", tokenMappingItem.apigeeAccessToken],
+      ["apigeeRefreshToken", tokenMappingItem.apigeeRefreshToken],
+      ["apigeeExpiresIn", expiryTimestamp],
+      ["selectedRoleId", tokenMappingItem.selectedRoleId],
+      ["userDetails", tokenMappingItem.userDetails],
+      ["rolesWithAccess", tokenMappingItem.rolesWithAccess],
+      ["rolesWithoutAccess", tokenMappingItem.rolesWithoutAccess],
+      ["currentlySelectedRole", tokenMappingItem.currentlySelectedRole]
+    ]
+
+    for (const [key, value] of optionalFields) {
+      if (value !== undefined && value !== null) {
+        expressionParts.push(`${key} = :${key}`)
+        expressionAttributeValues[`:${key}`] = value
+      }
+    }
+
+    const updateExpression = "SET " + expressionParts.join(", ")
+
+    await documentClient.send(
```

Comment:
I don't remember off the top of my head - but does this throw an error if the update fails? In the delete function below, we inspect the response status code to check it works, and if his doesn't throw an error we should do the same here. 

Similar logic for the put function above

(yes, I am aware I'm probably the one who wrote the original code for this...)